design/api.md

```diff
@@ -0,0 +1,219 @@
+The UCAN's audience needs to be `did:key:<account>`.
+
+**Response**:
+
+```ts
+{
+  email: string,
+  did: string,
+  username: string,
+}
+```
+
+---
+
+This fetches the latest account information.
+
+### GET `/api/v0/account/:username/did`
+
+---
+
+**Authentication**: *Unauthenticated*
+
+**Response**: `{ did: string }`
+
+---
+
+Use this to find out an account's DID. This information may be substituted by DoH & a DNS record in the future.
+
+
+
+
+### PUT `/api/v0/account/:did/username/:username`
+
+---
+
+**Authentication**:
+
+UCAN giving access to the capability `"fission:did:key:<account>": { "account/manage": [{}] }` originating in `did:key:<account>`.
+The UCAN's audience needs to be `did:key:<account>`.
+
+**Response**:
+
+Status 200 OK and `{ success: true }`, if successful.
+
+Status 409 Conflict and `{ success: false }`, if the username is already taken.
+
+---
+
+Change the account's username.
+
+
+### POST `/api/v0/fetch-capabilities/:did`
+
+---
+
+**Authentication**:
+
+```json
+{
+  "iss": "did:<from the path>",
+  "aud": "did:<server DID>",
+  "cap": {}, // Note: Should we introduce a capability? "did:key:zABC": { "capability/fetch": [{}] }?
+  "nnc": "<randomly generated>"
+}
+```
+
+**Request**: `{ ucans: string[] }`
+
+**Response**: `{ ucans: string[] }`
+
+---
+
+This returns a set of UCANs, that, combined with the UCAN from the request body will give the given DID access to new capabilities.
+
+In some cases this can be useful for logging in a new devices. E.g. when passkeys are available and are synced between devices, e.g. it's easy to ask a synced passkey to sign a UCAN giving the `"ucan:*": { "*": [{}] }` capability. However, this UCAN on its own doesn't give the account access to anything yet. It can use this endpoint to fetch the `fission:did:key:<account>` capability UCANs that delegate to the passkey.
+
+In the future this endpoint will be useful for fetching new capabilities from other services that were delegated to a user's account.
+
+
+## UCAN Capabilities
+
+### Resources
+
+The fission server understands one main resource, which is scoped to the `fission` URL scheme.
+
+Examples:
+- `fission:did:key:z6Mkw5d3acoQqn97UjRcGit7J7uWunixxxDKTgkr58CFHLfo`
+- `fission:did:key:z6Mkh5xiFmvPkWeHYZYnBLS4jVjsCjtpBS7D4thq1stdyEBK`
+- `fission:*`
+
+The URI's path identifies an account by DID. This DID is unique for every account and after delegating rights to the main server DID, the associated private key is deleted.
+
+The `fission:*` scheme subsumes any other resource schemes. It allows you to delegate account capabilities in cases where you don't know the account DIDs (e.g. when they're not generated yet).
+
+### Abilities
+
+```mermaid
+flowchart TD
+    Star["account/*"] --> Noncrit["account/noncritical"]
+    Noncrit --> Info["account/info"]
+    Star --> Manage["account/manage"]
+    Star --> Delete["account/delete"]
+```
+
+#### `account/noncritical` & `account/*`
```

Comment:
I like this distinction. It's a _bit_ like roles: essentially "query vs mutation vs dangerzone".

Reply:
(I may borrow this for spec examples)

Reply:
Yeah. This came from a discussion with Hugo. The idea being when you have a passkey, that passkey has critical access to your account, but delegates non-critical access to your session DID.
This means that if you want to delete or manage your account, you'll get another biometrics check before you do so.
It's also flexible: If we later decide we don't want to do a biometrics check for critical stuff *all the time*, but instead want to simply have a "critical account management area", then you'd simply delegate critical resources for a limited time once when entering the account management settings UI, and revoke after leaving.